### DISCUSSION TITLE
WordPress plugin

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#159] [Anything Else]
"WordPress is the largest CMS in the world with some 35% of all sites being on it. Could be nice with a simple WordPress integration: Plausible stats directly in the WordPress dashboard."

([imported from the old roadmap](https://plausible.nolt.io/123))

**Comment by binaryfire** [comment]
+1. This would be huge!

**Comment by boogah** [comment]
Yes please!

**Comment by danyork** [comment]
+1 to this request. My most critical requirement is an easy way to add the Plausible script into the headers of a WordPress site. I'd like to try out Plausible on a couple of my organization's smaller sites, but we use "default" WordPress themes on those sites that we do not modify. I therefore need a plugin to inject the the script into the header of the site. (We do something similar with GA right now, using a plugin to insert the GA javascript tracking script.)

I actually don't care as much about seeing the Plausible stats in the WordPress dashboard. I mean, that's cool and useful, but I can also just login to the Plausible website to get it.

If you do go ahead with this work, I'd be glad to help with testing.

**Comment by metmarkosaric** [comment]
Thanks for your feedback! It's something we'd definitely like to support. I use WordPress myself and add Plausible by editing my theme file currently.

If you need an immediate solution and don't need Plausible stats displayed in your WordPress dashboard, there are well supported plugins such as [this one](https://wordpress.org/plugins/insert-headers-and-footers/) with more than 1 million active users and created by Syed Balkhi and the WPBeginner team. It allows you to easily insert any code into your header without editing theme files. You could use this plugin to insert GA, Plausible and any other script.

Hope that helps, thanks!

**Comment by danyork** [comment]
@metmarkosaric - Aha! **Of course** there would be a WordPress plugin for inserting headers! Thanks for pointing that one out. I've got that installed now on two very small sites that are now sending stats to your service. Thanks!

**Comment by metmarkosaric** [comment]
Nice! There's a WordPress plugin for anything :smiley: 

Looking forward to hearing what you think @danyork!

**Comment by metmarkosaric** [comment]
Just published the instructions to our Docs in case someone else wants to insert Plausible Analytics code into their WordPress site either by [using a plugin or by modifying a theme file](https://docs.plausible.io/wordpress-integration/).

**Comment by binaryfire** [comment]
@metmarkosaric This is cool but I think it should be tracked in a separate issue? This issue is about a plugin for displaying stats in the dashboard (eg. via a REST API)

**Comment by metmarkosaric** [comment]
Thanks @binaryfire. Yeah, I just added it here for information as several people found this thread when looking for a simple WordPress integration. The thread will now focus on the plugin option that displays all stats within the WP dashboard.

**Comment by rje3** [comment]
It would be great to have a Plausible WP Dashboard widget as part of the WP plugin. 

I hacked one together using[ instructions here](https://codex.wordpress.org/Example_Dashboard_Widget) with the "[embed dashboard](https://plausible.io/docs/shared-links)" iframe feature in Plausible. 

The safest option is to use a WP plugin to customise the Admin Dashboard, create a HTML widget and to add the Plausible iframe info. All care no responsibility - back up your Wordpress site first if you try this below. 

1. Go to your Plausible Site settings > Visibility > create a Shared link. Scroll down and in "Embed dashboard" make and copy the iframe info.

2. Make a folder called "Plausible_Analytics_Widget" on your computer.

3. Add the attached file "plausible-dashboard-widget.php" :
[plausible-dashboard-widget.php.txt](https://github.com/plausible/analytics/files/6310371/plausible-dashboard-widget.php.txt)

Add and edit the "widget.php" file:
[widget.php.txt](https://github.com/plausible/analytics/files/6310374/widget.php.txt)

Change "insert Plausible iFrame info here" to the text copied from your Plausible embed dashboard in step 1.
<iframe plausible-embed src="insert Plausible iFrame info here" scrolling="yes" frameborder="0" loading="lazy" style="width: 1px; min-width: 100%; height: 600px;"></iframe>

4. Delete the .txt extension on the files to change them to php files (I couldn't upload them as php here).

5. Via FTP upload the "Plausible_Analytics_Widget" folder containing the "plausible-dashboard-widget.php" and "widget.php" files to your Wordpress plugin directory.

6. Activate the "Plausible_Analytics_Widget" from your Admin Dashboard Screen Options.


FWIW I made the Plausible Analytics Widget full width by adding the following to my Wordpress Child theme:

// force one-column dashboard
function shapeSpace_screen_layout_columns($columns) {
	$columns['dashboard'] = 1;
	return $columns;
}
add_filter('screen_layout_columns', 'shapeSpace_screen_layout_columns');

function shapeSpace_screen_layout_dashboard() { return 1; }
add_filter('get_user_option_screen_layout_dashboard', 'shapeSpace_screen_layout_dashboard');

**Comment by metmarkosaric** [comment]
New version of our WordPress plugin is out! And you can now check your stats directly in your WordPress admin dashboard! https://plausible.io/wordpress-analytics-plugin

**Reply by rje3** [reply]
Great job Marko and implemented so fast! Thank you.

**Reply by metmarkosaric** [reply]
all credit goes to @mehul0810

**Reply by sbw** [reply]
Excited to see this, but it seems to require PHP 8.0.  Two of my Wordpress sites went down when the plugin was updated, one automatically, the other when I updated the plugin manually.  (Not critical sites, so no biggie!)  I've removed the plugin from those sites for now.

Here's the error:

```
PHP Parse error:  syntax error, unexpected ')' in .../wp-content/plugins/plausible-analytics/src/Admin/Settings.php on line 204
```

That line is the closing parenthesis after a list of arguments to `sprintf()`.

See [Function Arguments](https://www.php.net/manual/en/functions.arguments.php): "As of PHP 8.0.0, the list of function arguments may include a trailing comma, which will be ignored."

I'm running Ubuntu 18.04, which installed PHP 7.2 by default.  I believe Wordpress doesn't support PHP 8.0 yet.

**Reply by metmarkosaric** [reply]
thanks for reporting that @sbw! there's another report here https://github.com/plausible/wordpress/issues/31

i'm running PHP 7.4 and no issues here. we'll see what's going on...

**Reply by metmarkosaric** [reply]
Version 1.2.1 has been deployed now with some fixes. Could you check now please @sbw?

**Reply by sbw** [reply]
@metmarkosaric, thanks!  I re-installed the plugin through the Wordpress plugins "Add" button.  The error is gone, and I can see my (non-admin) activity on plausible.io.

But when I go to "Analytics" on the Wordpress site's dashboard, I am prompted for a password.  That's even though I created a "Shared Link" with the password in the link.  Do I misunderstand how that works?  Must I also "Make stats publicly available"?

(I also had to disable Brave shields on my Wordpress site, since Brave interprets plausible.io as a tracker.  I wonder whether Brave has a mechanism for certifying analytics tools like Plausible as explicitly not a tracker.)

**Reply by metmarkosaric** [reply]
great, thanks for the update @sbw! you should create a shared link without a password. then you won't need to put in a password in WordPress either. 

Brave is a known issue as they're blocking privacy first analytics too. we're trying to start a dialogue and look for solutions but for now you'll need to disable the Brave shield on your own site

**Reply by sbw** [reply]
@metmarkosaric, sorry, I found the "shared link" thing confusing.  When I first went to "Visibility" in the Plausible settings, it displayed something that gave me the wrong impression.  That was only a few minutes ago, but I didn't make a note of what happened, so I can't give you a clear report.

I created a new shared link, and it works fine in the Wordpress dashboard now.

If I see the confusing display again with other sites, I'll make notes and grab screen shots.  Sorry for the fuzzy report.

**Reply by metmarkosaric** [reply]
sounds good, glad to hear it's all working now!

**Reply by sbw** [reply]
@metmarkosaric, I figured out in what way I was thick-headed.  It wasn't the "Visibility" page in Plausible that confused me, but the plugin settings page in my Wordpress site, where it says:

Shared link: `https://plausible.io/share/example.com?auth=------------`

I replaced the dashes with a new random password, then created a shared link in Plausible with that password.  I didn't even look at the shared link that Plausible created, because I thought I was already done with the configuration.

I don't know how you might make it clearer.  Or even whether you should try: I'm an outlier in many ways, including what confuses me.

Anyway, I wanted you to know what mistake I made.

**Reply by metmarkosaric** [reply]
thanks for the update!

**Comment by binaryfire** [comment]
@metmarkosaric @mehul0810 Nice work guys!

**Comment by rje3** [comment]
Now custom domains are depreciated is it possible to add Cloudflare proxy information?

**Reply by metmarkosaric** [reply]
We plan to build a proxy directly in the next major version of the plugin. You can follow the progress here https://github.com/plausible/wordpress/issues/23

**Reply by rje3** [reply]
Thank you Marko. :)

Meantime I've set up a Cloudflare proxy as per [your instructions](https://plausible.io/docs/proxy/guides/cloudflare).

**Reply by metmarkosaric** [reply]
yeah, i did the same on my blog. hopefully we'll see this in our plugin soon!
